modula3/README.md

```diff
@@ -2,18 +2,33 @@
 
 Author: John Perry (john.perry@usm.edu)
 
+## Versions
+
+There are three implementations of the code here.
+  * The standard version (in the `src` directory) is a naive implementation.
+    It uses Modula-3's garbage collection facilities.
+  * The first tuned version (in the `src_unsafe` directory) uses untraced references.
+    As such, it has to `DISPOSE()` of them itself, and this makes the thing unsafe.
+  * The second tuned version (in the `src_pooled` directory) uses a generic memory pool.
+    This requires the construction of an intermediate generic interface and module,
+    which is a little cumbersome, but that's what seemingly pointless `NodeMemoryPool` files are about.
+    Instead of relying on Modula-3's `NEW()` and `DISPOSE()` commands,
+    the implementation instead invokes the pool's `Create()` and `Dispose()` commands.
+
 ## Compile
 
 To compile with the [Critical Mass Modula-3 compiler](https://github.com/modula3/cm3/releases),
 first change to the version you wish to compile.
 Once you're in that directory, type
 
 ```
-cm3 -O
+cm3 -O3
```

Comment:
Which version of the compiler do you use? Critical Mass Modula-3 d5.10.0 does not accept `-O3`.

Reply:
I'm using d5.10.0 and it accepts `-O3`. I just tested it again to be sure. I'm not sure why, because I remember at one point it didn't on at least one of my machines (actually I think the one I'm trying now), and it annoyed me because the documentation says it did. However, it doesn't seem to make any difference in the speed, so it's probably best to switch back to `-O`.

Reply:
For what it's worth, here's what I get when I type, `cm3 -?`.

```
Critical Mass Modula-3 version d5.10.0
  last updated: 2016-10-05
  compiled: 2018-10-28 04:35:08
  configuration: /usr/local/cm3/bin/cm3.cfg
  host: AMD64_DARWIN
  target: AMD64_DARWIN
```

I notice that I recompiled it on this machine at one point. That may be what got `-O3` working, but as I say, `-O` seems to have the same time.